src/main/java/co/rsk/bitcoinj/core/StoredBlock.java

```diff
@@ -166,6 +176,22 @@ public static StoredBlock deserializeCompact(NetworkParameters params, ByteBuffe
         return new StoredBlock(params.getDefaultSerializer().makeBlock(header), chainWork, height);
     }
 
+    /**
+     * Deserializes the stored block from a custom packed format. Used internally.
+     *
+     * @param buffer data to deserialize
+     * @return deserialized stored block
+     */
+    public static StoredBlock deserializeCompactV2(NetworkParameters params, ByteBuffer buffer) throws ProtocolException {
```

Comment:
Same comment as with the serialize method. If both are equal except for the chain work size we could put all in a private method

Reply:
Done